sdk.go

```diff
@@ -151,11 +154,64 @@ func Serve(fn v1beta1.FunctionRunnerServiceServer, o ...ServeOption) error {
 
 	srv := grpc.NewServer(grpc.MaxRecvMsgSize(so.MaxRecvMsgSize), grpc.Creds(so.Credentials))
 	reflection.Register(srv)
-	v1beta1.RegisterFunctionRunnerServiceServer(srv, fn)
+	v1.RegisterFunctionRunnerServiceServer(srv, fn)
+	v1beta1.RegisterFunctionRunnerServiceServer(srv, ServeBeta(fn))
 	return errors.Wrap(srv.Serve(lis), "cannot serve mTLS gRPC connections")
 }
 
 // NewLogger returns a new logger.
 func NewLogger(debug bool) (logging.Logger, error) {
 	return logging.NewLogger(debug)
 }
+
+// A BetaServer is a v1beta1 FunctionRunnerServiceServer that wraps an identical
+// v1 FunctionRunnerServiceServer. This requires the v1 and v1beta1 protos to be
+// identical.
+//
+// Functions were promoted from v1beta1 to v1 in Crossplane v1.17. Crossplane
+// v1.16 and earlier only sends v1beta1 RunFunctionRequests. Functions should
+// use the BetaServer for backward compatibility, to support Crossplane v1.16
+// and earlier.
```

Comment:
This is the most interesting part of this PR. Everything else is just:

* Replicating the v1 and v1beta1 protos from Crossplane
* `s/v1beta1/v1/` for all the imports of the protos